src/morale.cpp

```diff
@@ -357,66 +357,85 @@ void player_morale::decay( int ticks )
-                        ( TERMX > FULL_SCREEN_WIDTH ) ? ( TERMX - FULL_SCREEN_WIDTH ) / 2 : 0 );
-    draw_border( w );
+    const char *morale_gain_caption = _( "Total morale gain" );
+    const char *focus_gain_caption = _( "Focus gain per minute" );
 
-    // Figure out how wide the name column needs to be.
-    int name_column_width = 18;
+    // Figure out how wide the source column needs to be.
+    int source_column_width = utf8_width( focus_gain_caption );
+    if( utf8_width( morale_gain_caption ) > source_column_width ) {
+        source_column_width = utf8_width( morale_gain_caption );
+    }
     for( auto &i : points ) {
-        int length = utf8_width( i.get_name() );
-        if( length > name_column_width ) {
-            name_column_width = length;
-            // If it's too wide, truncate.
-            if( name_column_width >= 72 ) {
-                name_column_width = 72;
-                break;
-            }
+        const int length = utf8_width( i.get_name() );
+        if( length > source_column_width ) {
+            source_column_width = length;
         }
     }
 
-    // Header
-    mvwprintz( w, 1,  1, c_white, _( "Morale Modifiers:" ) );
-    mvwprintz( w, 2,  1, c_ltgray, _( "Name" ) );
-    mvwprintz( w, 2, name_column_width + 2, c_ltgray, _( "Value" ) );
+    const int win_w = std::min( source_column_width + 4 + 8, FULL_SCREEN_WIDTH );
+    const int win_h = FULL_SCREEN_HEIGHT;
+    const int win_x = ( TERMX - win_w ) / 2;
+    const int win_y = ( TERMY - win_h ) / 2;
 
-    // Start printing the number right after the name column.
-    // We'll right-justify it later.
-    int number_pos = name_column_width + 1;
+    WINDOW *w = newwin( win_h, win_w, win_y, win_x );
 
-    const morale_mult mult = get_temper_mult();
-    // Print out the morale entries.
-    for( size_t i = 0; i < points.size(); i++ ) {
-        const std::string name = points[i].get_name();
-        const int bonus = points[i].get_net_bonus( mult );
-        const nc_color bonus_color = ( bonus < 0 ? c_red : c_green );
+    draw_border( w );
 
-        // Print out the name.
-        trim_and_print( w, i + 3,  1, name_column_width, bonus_color, name.c_str() );
+    mvwprintz( w, 1, 2, c_white, _( "Morale" ) );
+
+    mvwhline( w, 2, 0, LINE_XXXO, 1 );
+    mvwhline( w, 2, 1, 0, win_w - 2 );
+    mvwhline( w, 2, win_w - 1, LINE_XOXX, 1 );
+
+    mvwhline( w, win_h - 4, 0, LINE_XXXO, 1 );
+    mvwhline( w, win_h - 4, 1, 0, win_w - 2 );
+    mvwhline( w, win_h - 4, win_w - 1, LINE_XOXX, 1 );
+
+    const auto print_line = [ w ]( int y, const char *label, double value ) -> int {
+        nc_color color;
+        if( value != 0.0 )
+        {
+            const int decimals = ( value - static_cast<int>( value ) != 0.0 ) ? 2 : 0;
+            color = ( value > 0.0 ) ? c_green : c_red;
+            mvwprintz( w, y, getmaxx( w ) - 8, color, "%+6.*f", decimals, value );
+        } else
+        {
```

Comment:
Please read doc/CODE_STYLE.txt and apply the style to the code (example: `} else {` on one line).